workflow/task_manager.py

```diff
@@ -424,6 +426,357 @@ def _slurm_batch_submit(command,
+    #---- end slurm directives
+
+    batch_script_pre.extend([
+        'if [ -z $MODULEPATH_ROOT ]; then',
+        '  unset MODULEPATH_ROOT',
+        'else',
+        '  echo "NO MODULEPATH_ROOT TO RESET"',
+        'fi',
+        'if [ -z $MODULEPATH ]; then',
+        '  unset MODULEPATH',
+        'else',
+        '  echo "NO MODULEPATH TO RESET"',
+        'fi',
+        'if [ -z $LMOD_SYSTEM_DEFAULT_MODULES ]; then',
+        '  unset LMOD_SYSTEM_DEFAULT_MODULES',
+        'else',
+        '  echo "NO LMOD_SYSTEM_DEFAULT_MODULES TO RESET"',
+        'fi',
+        'source /etc/profile',
+        'export TERM='+env['TERM'],
+        'export HOME='+env['HOME']])
+
+    #-- I get the following error in Python reading netcdf4 files:
+    # ImportError: /lib64/libk5crypto.so.3: symbol krb5int_buf_len, version
+    #   krb5support_0_MIT not defined in file libkrb5support.so.0 with link time
+    #   reference
+    # this fixes it:
+    batch_script_pre.append('unset LD_LIBRARY_PATH')
+
+    batch_script_pre.append('export {CONDA_PATH}:$PATH'.format(CONDA_PATH=CONDA_PATH))
+    batch_script_pre.append('export PYTHONUNBUFFERED=False')
+    batch_script_pre.append('export TMPDIR='+TMPDIR)
+
+    if module_purge:
+        batch_script_pre.append('module purge')
+
+    if modules:
+        for module in modules:
+            batch_script_pre.append('module load '+module)
+        batch_script_pre.append('module list')
+
+    if conda_env:
+        batch_script_pre.append('source activate %s'%conda_env)
+
+    batch_script_post = ['exit ${?}']
+
+    #-- write run script
+    with open(batch_script_file,'w') as fid:
+        for line in batch_script_pre+cmd_line+batch_script_post:
+            fid.write('%s\n'%line)
+
+    #-- make run script executable
+    st = os.stat(batch_script_file)
+    os.chmod(batch_script_file, st.st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH )
+
+    #-- submit the job
+    p = Popen(['qsub',batch_script_file],
+              stdin=None,
+              stdout=PIPE,
+              stderr=PIPE,
+              env=env)
+    stdout, stderr = p.communicate()
+
+    stdout = stdout.decode('UTF-8')
+    stderr = stderr.decode('UTF-8')
+
+    #-- parse return string to get job ID
+    try:
+        # Trying to figure out jobid
+        print(stdout)
+        jid = stdout.splitlines()[-1].split(' ')[-1].strip()
+        JID.append(jid)
+    except:
+        print('SLURM sbatch failed!')
+        print('Command:')
+        print(command)
+        print('\nstdout:')
+        print(stdout)
+        print('\nstderr:')
+        print(stderr)
+        raise
+
+    #-- print job id and job submission string
+    scmd = '; '.join(cmd_line)
+    print('-'*50)
+    print('%s (%s): %s'%(jid,os.path.basename(batch_script_file),scmd))
+    print('-'*50)
+    print()
+
+    if total_elapsed_time() > QUEUE_MAX_HOURS:
+        stop = True
+
+    #-- return job id
+    return jid,ok,stop
+
+def _slurm_batch_submit(command,
```

Comment:
Is this a duplicate copy of an existing slurm function? I'm surprised something slurm-related is showing up as an addition.

Reply:
It is a duplicate -  I removed it!